samples/apps/copilot-chat-app/webapi/Services/AppInsightsTelemetryService.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Security.Claims;
+using Microsoft.ApplicationInsights;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace SemanticKernel.Service.Services;
+
+/// <summary>
+/// Implementation of the telemetry service interface for Azure Application Insights (AppInsights).
+/// </summary>
+public class AppInsightsTelemetryService : ITelemetryService
+{
+    private readonly TelemetryClient _telemetryClient;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    /// <summary>
+    /// Creates an instance of the app insights telemetry service.
+    /// This should be injected into the service collection during startup.
+    /// </summary>
+    /// <param name="telemetryClient">An AppInsights telemetry client</param>
+    /// <param name="httpContextAccessor">Accessor for the current request's http context</param>
+    public AppInsightsTelemetryService(TelemetryClient telemetryClient, IHttpContextAccessor httpContextAccessor)
+    {
+        this._telemetryClient = telemetryClient;
+        this._httpContextAccessor = httpContextAccessor;
+    }
+
+    /// <inheritdoc/>
+    public void TrackSkillEvent(string skillName, string functionName, bool success)
```

Comment:
From line 39, we see this is for completion of a skill function. Maybe rename this method to TrackFunctionCompletion?

Reply:
TrackSkillFunction perhaps?